pennylane/math/multi_dispatch.py

```diff
@@ -54,64 +53,6 @@ def eye(*args, like=None, **kwargs):
     return res
 
 
-def _multi_dispatch(values):
-    """Determines the correct framework to dispatch to given a
-    sequence of tensor-like objects.
-
-    Args:
-        values (Sequence[tensor_like]): a sequence of tensor like objects
-
-    Returns:
-        str: the name of the interface
-
-    To determine the framework to dispatch to, the following rules
-    are applied:
-
-    * Tensors that are incompatible (such as Torch and TensorFlow tensors)
-      cannot both be present.
-
-    * Autograd tensors *may* be present alongside Torch and TensorFlow tensors,
-      but Torch and TensorFlow take precendence; the autograd arrays will
-      be treated as non-differentiable NumPy arrays. A warning will be raised
-      suggesting that vanilla NumPy be used instead.
-
-    * Vanilla NumPy arrays and SciPy sparse matrices can be used alongside other tensor objects;
-      they will always be treated as non-differentiable constants.
-    """
-    if "resource_variable" in getattr(values, "__module__", tuple()):
-        values = np.asarray(values)

```

Comment:
This logic seems to have been removed completely. How come?

Reply:
It became dead code after the refactors

Reply:
Why?

Reply:
I have no idea, CodeCov says these lines were covered before, but I created a test PR deleting these lines and all the tests passed...

@josh146 I see you edited these lines 17 months ago. Do you remember the reason for doing this?

Reply:
@AlbertMitjans I don't recall, sorry 😬 Although it looks like I was trying to catch some tensorflow specific edgecase?

![image](https://user-images.githubusercontent.com/2959003/195135394-423a2c71-90da-4615-b3c8-ecd449fa57dd.png)

Reply:
I see, thanks! For some reason you casted all tensorflow Variables to numpy arrays...

Given that these lines of code were located in a private function and apparently the tests don't rely on them, do you think we can delete them?

Reply:
It is an iterable, but maybe this is a new addition:

```pycon
>>> var = tf.Variable([0, 1, 2])
>>> for val in var:
    print(var)
tf.Tensor(0, shape=(), dtype=int32)
tf.Tensor(1, shape=(), dtype=int32)
tf.Tensor(2, shape=(), dtype=int32)
```

Reply:
Interesting. It may also be that the TF version used at the time had some behaviour for which this was required.

This is good to go from my side, thanks for more details here!

Reply:
Oh this is great to know!! This must be a recent change, it would be good to find out which TF version made this change.

If you are going through the PL code base and see weird things like:

- TF Variables multiplied by 1.0 (e.g., https://github.com/PennyLaneAI/pennylane/blob/32fb63bb988b70cfd2a5ed770a89f8fcf5529d89/pennylane/qnn/keras.py#L318)
- TF Variables seemingly redundantly converted to tensors via `tf.convert_to_tensor()`

this is because we have been converting `tf.Variable` (not iterable) -> `tf.Tensor` (iterable) to get around this weird behaviour.

Reply:
https://github.com/tensorflow/tensorflow/issues/23185 Looks like this was fixed in Sept 2021